daemon/cmd/cni/config.go

```diff
@@ -42,29 +42,35 @@ type cniConfigManager struct {
 
 	// watcher watches for changes in the CNI configuration directory
 	watcher *fsnotify.Watcher
+
+	// mtu obtained from CNI Configuration file
+	mtu int
 }
 
 // GetMTU returns the MTU as written in the CNI configuration file.
 // This is one way to override the node's MTU.
 func (c *cniConfigManager) GetMTU() int {
+	return c.mtu
+}
+
+// GetChainingMode returns the configured chaining mode.
+func (c *cniConfigManager) GetChainingMode() string {
+	return c.config.CNIChainingMode
+}
+
+// GetNetConf returns the *NetConf obtained from CNI configuration file
+func (c *cniConfigManager) GetNetConf() *cnitypes.NetConf {
 	if c.config.ReadCNIConf == "" {
-		return 0
+		return nil
 	}
 
 	conf, err := cnitypes.ReadNetConf(c.config.ReadCNIConf)
 	if err != nil {
 		c.log.WithField("path", c.config.ReadCNIConf).WithError(err).Warnf("Failed to parse existing CNI configuration file")
 	}
```

Comment:
there is a bug here in the existing codebase; this should return 0 here as well.